applicationset/services/internal/http/client.go

```diff
@@ -102,7 +102,7 @@ func (c *Client) NewRequest(method, path string, body any, options []ClientOptio
 	return req, nil
 }
 
-func (c *Client) Do(ctx context.Context, req *http.Request, v any) (*http.Response, error) {
+func (c *Client) Do(_ context.Context, req *http.Request, v any) (*http.Response, error) {
```

Comment:
The only non-test caller of `Do` is `List`. Let's remove the param from `Do` and in `List` use `NewRequestWithContext` instead of `NewRequest`.

I know it ends up using `context.Background()` at the top of the call stack, but this will future-proof in case we ever use a proper context instead.